### PR TITLE
Delivery/feature/login logic

### PR DESCRIPTION
**Pull request by AzizaHelmy**
- finish login integration
- finish permission intergartion
- save tokens in db
- add interceptor
- and refactor some code

### CODE REVIEW

File: client_delivery/shared/src/commonMain/kotlin/presentation/base/BaseScreenModel.kt [hunk 1]
```diff
@@ -71,7 +75,16 @@ abstract class BaseScreenModel<S, E>(initialState: S) : ScreenModel, KoinCompone
...
+            } catch (e: InvalidUsernameException) {
+                onError(ErrorState.InvalidUsername)
+            } catch (e: InvalidPasswordException) {
+                onError(ErrorState.InvalidPassword)
+            } catch (e: UserNotFoundException) {
+                onError(ErrorState.UserNotFound)
+            } catch (e: UnAuthorizedException) {
+                onError(ErrorState.UnAuthorized)
             } catch (e: Exception) {
+                println("mohamed: $e")
```
Comment (mosayed01) [inline client_delivery/shared/src/commonMain/kotlin/presentation/base/BaseScreenModel.kt:87]: u can clone the way we handled exceptions in this PR #364 
may in another PR but that way is better.

Comment (mosayed01) [inline client_delivery/shared/src/commonMain/kotlin/presentation/base/BaseScreenModel.kt:87]:
```suggestion
```

File: client_delivery/shared/src/commonMain/kotlin/presentation/login/LoginScreenModel.kt [hunk 2]
```diff
@@ -1,18 +1,29 @@
...
 
     override val viewModelScope: CoroutineScope
         get() = coroutineScope
 
+    init {
+        viewModelScope.launch {
+            if (manageLoginUser.getKeepMeLoggedInFlag()) {
+                sendNewEffect(LoginScreenUIEffect.LoginEffect(""))
+            }
+        }
```
Comment (Asiasama710) [inline client_delivery/shared/src/commonMain/kotlin/presentation/login/LoginScreenModel.kt:24]: move these line into function

**Review by Asiasama710 — APPROVED** [review]

**Review by Youssef-Ramadan-Y — APPROVED** [review]

**Review by Hassan3Ayman — APPROVED** [review]

File: client_delivery/shared/src/commonMain/kotlin/data/gateway/remote/AuthorizationInterceptor.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import io.ktor.client.plugins.plugin
+import io.ktor.client.request.headers
+import org.koin.core.scope.Scope
+
+fun Scope.authorizationIntercept(client: HttpClient) {
+
+    client.plugin(HttpSend).intercept { request ->
+
+        val localConfigurationGateway = get<LocalConfigurationGateway>()
+        // todo add identity gate way to fix this error
```
Comment (mosayed01) [inline client_delivery/shared/src/commonMain/kotlin/data/gateway/remote/AuthorizationInterceptor.kt:15]: still need this todo?

**Review by mosayed01 — APPROVED** [review]

**Pull request merged**